packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js

```diff
@@ -37,22 +38,27 @@ export default {
     ],
   },
   create(context) {
+    const inputOptions = context.options && context.options[0];
     // Parse the `additionalHooks` regex.
     const additionalHooks =
-      context.options &&
-      context.options[0] &&
-      context.options[0].additionalHooks
-        ? new RegExp(context.options[0].additionalHooks)
+      inputOptions && inputOptions.additionalHooks
+        ? new RegExp(inputOptions.additionalHooks)
+        : undefined;
+
+    // Parse the `safeHooks` regex
+    const safeHooks =
+      inputOptions && inputOptions.safeHooks
```

Comment:
nit-pick: A bit of duplication here could be refactored into:

```
function getHookType(inputOptions, type) {
  return inputOptions && inputOptions[type]
        ? new RegExp(inputOptions[type])
        : undefined;
}
```

Then we can call:
```
const additionalHooks = getHookType(inputOptions, 'additionalHooks');
const safeHooks = getHookType(inputOptions, 'safeHooks');
```

Reply:
Personally I don't think it's worth the abstraction, since the duplication is so minimal (and it's only lexical duplication, no unnecessary runtime duplication). I think adding the function will just add a bit more overhead when trying to read the code cause you have to figure out what that helper func does.